cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java

```diff
@@ -317,41 +337,36 @@ private boolean updateEndpointResources(String linkFormat) {
 	}
 
 	/*
-	 * the following three methods are used to print the right string to put in
-	 * the payload to respond to the GET request.
+	 * the following two methods put the resources of this node
+	 * in a list of strings in link format. Each string ends with ','.
 	 */
-	public String toLinkFormat(List<String> query) {
-
-		// Create new StringBuilder
-		StringBuilder builder = new StringBuilder();
-		
-		// Build the link format
-		buildLinkFormat(this, builder, query);
-
-		// Remove last delimiter
-		if (builder.length() > 0) {
-			builder.deleteCharAt(builder.length() - 1);
-		}
-
-		return builder.toString();
+	public List<String> toLinkFormat(List<String> query) {
+		// Build the link format recursively
+		return buildLinkFormat(this, query);
 	}
 
-	private void buildLinkFormat(Resource resource, StringBuilder builder, List<String> query) {
+	private List<String> buildLinkFormat(Resource resource, List<String> query) {
+		List<String> resources = new ArrayList<>();
 		if (resource.getChildren().size() > 0) {
 
 			// Loop over all sub-resources
 			for (Resource res : resource.getChildren()) {
+				String link = "";
```

Comment:
I would appreciate, if `StringBuilder` gets used.

Reply:
Changed to StringBuilder in fixup commit.